python/cudf_polars/cudf_polars/utils/dtypes.py

```diff
@@ -84,8 +86,16 @@ def can_cast(from_: plc.DataType, to: plc.DataType) -> bool:
                 )
             )
         )
-        or (from_.id() == plc.TypeId.STRING and is_numeric_not_bool(to))
-        or (to.id() == plc.TypeId.STRING and is_numeric_not_bool(from_))
+        or (
+            from_.id() == plc.TypeId.STRING
+            and not to_is_empty
+            and is_numeric_not_bool(to)
+        )
+        or (
+            to.id() == plc.TypeId.STRING
+            and not from_is_empty
+            and is_numeric_not_bool(from_)
```

Comment:
I would assume `is_numeric_not_bool` would return `False` for `plc.TypeId.EMPTY`?

Or were you observing this behavior?

```python
In [4]: pylibcudf.traits.is_numeric_not_bool(pylibcudf.DataType(pylibcudf.TypeId.EMPTY))
terminate called after throwing an instance of 'cudf::logic_error'
  what():  CUDF failure at:/cpp/include/cudf/utilities/type_dispatcher.hpp:546: Invalid type_id.
Aborted (core dumped)
```

Reply:
Yes, I was observing that error. I think this may not be too hard to support. I think we just need to add case `type_id::EMPTY` case to `type_dispatcher`. I'll try an open a PR

Reply:
I opened https://github.com/rapidsai/cudf/pull/18109. If its accepted, I can revert the changes to `can_cast` in this PR

Reply:
Great thanks! This review is non-blocking so approved anyways.